tutorials/s4sdk-resilience/s4sdk-resilience.md

```diff
@@ -37,100 +37,132 @@ In order to avoid such scenarios, we need to equip applications with the ability
+1. The code we want to execute in a resilient manner. It can be either a `Supplier`, `Callable`, `Supplier<Future>`, method reference, or a simple lambda function. As you might have noticed already, we have simply taken the VDM-based code that calls our OData service from the previous tutorial, and put it into a separate run() method. The `ResilienceDecorator` offers methods that simply wrap the provided function and returns a new function (`decorateSupplier`, `decorateCallable`, etc.), plus methods that also start execution the function immediately (`executeSupplier`, `executeCallable`, etc.). Here we used `executeSupplier` with a method reference to the VDM-based code.
 
-In the following example we set the thread-pool size to 20 and the timeout to 10000 milliseconds.
+2. An instance of `ResilienceConfiguration` with identifier parameter set. Here we use the class reference, but a string identifier can also be used. Besides the mandatory identifier parameter, the SAP Cloud SDK comes with a default resilience configuration, so you don't need to perform any other configuration on your own. In most cases the default configuration will suffice. However, if you need to change the resilience configuration, you can find more information on this topic in [SAP Cloud SDK Javadoc](https://help.sap.com/doc/ae45330c443b42c5a54bde85dd70aec9/1.0/en-US/com/sap/cloud/sdk/cloudplatform/resilience/ResilienceConfiguration.html)
 
+Here is an example of a custom resilience configuration. Here we set the isolation mode to optional tenant + user, the bulkhead maximum concurrent calls to 20, and the execution timeout to 10000 milliseconds.
+
+```java
+myResilienceConfig = ResilienceConfiguration.of(BusinessPartnerService.class)
+        .isolationMode(ResilienceIsolationMode.TENANT_AND_USER_OPTIONAL)
+        .timeLimiterConfiguration(
+                ResilienceConfiguration.TimeLimiterConfiguration.of()
+                        .timeoutDuration(Duration.ofMillis(10000)))
+        .bulkheadConfiguration(
+                ResilienceConfiguration.BulkheadConfiguration.of()
+                        .maxConcurrentCalls(20));
 ```
-protected GetBusinessPartnersCommand() {
-    super(HystrixUtil
-            .getDefaultErpCommandSetter(
-                    GetBusinessPartnersCommand.class,
-                    HystrixUtil.getDefaultErpCommandProperties().withExecutionTimeoutInMilliseconds(10000))
-            .andThreadPoolPropertiesDefaults(HystrixThreadPoolProperties.Setter().withCoreSize(20)));
-}
-```
-Now that we have a working command, we need to adapt our `BusinessPartnerServlet`:
+
+Additionally, the `decorate...` and `execute...` methods of `ResilienceDecorator` support an optional third parameter for a fallback function, in case the remote service call should fail. In this case we have a lambda function that returns an empty list. We could also serve static data or check whether we have already cached a response to this call.
```

Comment:
"Best practices is to at least log the provided Throwable."

Reply:
Updated